### PR TITLE
KAFKA-7584: StreamsConfig throws ClassCastException if max.in.flight.request.per.connect is specified as String

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @guozhangwang @bbejeck @vvcephei

**Comment by mjsax** [comment]
Java11 issues again. Failing Streams tests timed out. Additionally:
```
java.lang.AssertionError: 
Expected: <[KeyValue(0, 0), KeyValue(0, 1), KeyValue(0, 3), KeyValue(0, 6), KeyValue(0, 10), KeyValue(0, 15), KeyValue(0, 21), KeyValue(0, 28), KeyValue(0, 36), KeyValue(0, 45), KeyValue(0, 55), KeyValue(0, 66), KeyValue(0, 78), KeyValue(0, 91), KeyValue(0, 105), KeyValue(0, 120), KeyValue(0, 136), KeyValue(0, 153), KeyValue(0, 171), KeyValue(0, 190)]>
     but: was <[KeyValue(0, 0), KeyValue(0, 1), KeyValue(0, 3), KeyValue(0, 6), KeyValue(0, 10), KeyValue(0, 15), KeyValue(0, 21), KeyValue(0, 28), KeyValue(0, 36), KeyValue(0, 45), KeyValue(0, 10), KeyValue(0, 21), KeyValue(0, 33), KeyValue(0, 46), KeyValue(0, 60), KeyValue(0, 75), KeyValue(0, 91), KeyValue(0, 108), KeyValue(0, 126), KeyValue(0, 145)]>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
	at org.apache.kafka.streams.integration.EosIntegrationTest.checkResultPerKey(EosIntegrationTest.java:218)
	at org.apache.kafka.streams.integration.EosIntegrationTest.shouldNotViolateEosIfOneTaskFailsWithState(EosIntegrationTest.java:467)
```
Also some non-Streams test failures.

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 1]
```diff
@@ -887,10 +887,30 @@ private void checkIfUnexpectedUserSpecifiedConsumerConfig(final Map<String, Obje
...
+                    } catch (final NumberFormatException e) {
+                        throw new ConfigException("Config value for " + ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION + " could not be parsed into an integer");
+                    }
+                } else if (maxInFlightRequests instanceof Number) {
+                    maxInFlightRequestsAsInteger = ((Number) maxInFlightRequests).intValue();
+                } else {
+                    throw new ConfigException("Config value (with type " + maxInFlightRequests.getClass().getName() + " for " + ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION + " could not be converted into an int.");
+                }
+
+                if (5 < maxInFlightRequestsAsInteger) {
```
Comment (bbejeck) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: super nit: maybe reverse statement  to `maxInFlightRequestsAsInteger > 5` IMHO easier to grok, but this is highly opinionated, so feel free to ignore.

**Review by bbejeck — APPROVED** [review]
Thanks @mjsax just one minor nit otherwise LGTM

**Comment by mjsax** [comment]
Failing Java11 tests unrelated:
```
kafka.api.ClientIdQuotaTest.testThrottledProducerConsumer
kafka.api.ClientIdQuotaTest.testQuotaOverrideDelete
kafka.api.UserClientIdQuotaTest.testThrottledProducerConsumer
kafka.server.DynamicBrokerReconfigurationTest.testUncleanLeaderElectionEnable
```

**Comment by mjsax** [comment]
Updated this.

**Comment by bbejeck** [comment]
failures unrelated 

retest this please

**Review by kamalcph — COMMENTED** [review]
LGTM.

**Comment by mjsax** [comment]
Retest this please

**Comment by vvcephei** [comment]
LGTM. Thanks @mjsax !

**Review by guozhangwang — APPROVED** [review]

**Comment by mjsax** [comment]
Did you cherry-pick this to 2.0 and 2.1? (earlier version are not affected).

**Comment by guozhangwang** [comment]
Have not, thanks for reminding!

**Comment by guozhangwang** [comment]
Cherry-picked to 2.0 and 2.1 as well.

**Comment by mjsax** [comment]
Thank you!

**Pull request merged**